src/transformers/modeling_tf_utils.py

```diff
@@ -656,14 +665,16 @@ def generate(
             bad_words_ids is None or isinstance(bad_words_ids, list) and isinstance(bad_words_ids[0], list)
         ), "`bad_words_ids` is either `None` or a list of lists of tokens that should not be generated"
 
-        if input_ids is None:
+        if prefix_ids is None:
```

Comment:
I don't like `prefix_ids` overwriting `input_ids` as mentioned before. Why should we do that?